### DISCUSSION TITLE
notification lenght

### DISCUSSION THREAD

**Discussion by mantegna10** [ThomasGerstenberg/blatann#87] [Q&A] [answered]
Hi,
I'm trying to **receive a notification** (my dongle act as central device) from a characteristic at witch I subscribe.
Everithing works fine except for the fact that the length of the notication is too short.
I'm receiving a string **message that is truncated**. 
I've repeated the test with the _"Nordic nRF Connect Desktop" BLE app_ and in that way the message is complete.
Is there a way to change or set the maximum legth of a notification? 
Thanks.
Luca

**Comment by ThomasGerstenberg** [comment, marked as answer]
Hi Luca,
The notification length is determined by the connection's MTU. The default MTU size is 23 bytes, where notifications require 3 bytes overhead thus 20 bytes are notified. The MTU size is typically negotiated at the start of a connection and set to a larger size if desired.

My guess is nRF Connect does the MTU negotiation automatically whereas in blatann it's a manual function call. Typically it looks something like this:

```py
device = BleDevice(params)

 # This sets the maximum MTU size supported by the dongle. Without this call the max is set to the default 23. BLE spec is 512 bytes. 
# if configure() is called without any parameters, the max is set to 247 which is optimized for DLE/throughput
device.configure(att_mtu_max_size=x) 


peer = device.connect(...).wait()

print(f"Initial MTU size: {peer.mtu_size}") 
peer.preferred_mtu_size = desired_mtu_size  # Set the preferred MTU size for this peer
peer.exchange_mtu().wait()  # Perform MTU exchange. Both devices send their preferred MTU size and the smaller of the two is set
print(f"New MTU size: {peer.mtu_size}") 

peer.update_data_length().wait()  # Sets the DLE to be optimal for the new MTU, helps throughput

# perform service discovery and everything else
```

Hope this helps!

**Reply by mantegna10** [reply]
Thomas,
it totally **worked**. 
I've solved the problem and probably you are right also about nRF Connect.
Thanks for your support.
Best Reguards,
Luca
